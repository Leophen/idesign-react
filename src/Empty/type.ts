export interface EmptyProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 自定义展位图
   */
  image?: string;
  /**
   * 空状态尺寸
   * @default medium
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * 空状态显示类型
   * @default default
   */
  type?: 'default' | 'shoppingTrolley' | 'favorite' | 'gold';
}
