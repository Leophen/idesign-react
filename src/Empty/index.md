---
nav:
  title: 组件
  path: /components
group:
  title: 数据展示组件
  order: 4
order: 2
---

# Empty 空状态

空状态时的展示占位图。

## 基本用法

`Empty` 可直接包裹展位图提示文字。

```tsx
import React from 'react';
import { Empty } from 'idesign-react';

const App = () => {
  return <Empty>暂无数据</Empty>;
};

export default App;
```

## 不同尺寸

提供 `small`、`medium`（默认）、`large` 三种不同尺寸的空状态展位图。

```tsx
import React from 'react';
import { Empty } from 'idesign-react';

const App = () => {
  return (
    <>
      <Empty size="small">暂无数据</Empty>
      <Empty size="medium">暂无数据</Empty>
      <Empty size="large">暂无数据</Empty>
    </>
  );
};

export default App;
```

## 不同类型

提供 `default`（默认）、`shoppingTrolley`、`favorite`、`gold` 四种不同类型的空状态展位图。

```tsx
import React from 'react';
import { Empty } from 'idesign-react';

const App = () => {
  return (
    <>
      <Empty type="default">暂无数据</Empty>
      <Empty type="shoppingTrolley">购物车空空如也</Empty>
      <Empty type="favorite">收藏夹为空</Empty>
      <Empty type="gold">暂无金币</Empty>
    </>
  );
};

export default App;
```

## 自定义展位图

提供 `image` 属性自定义展位图。

```tsx
import React from 'react';
import { Empty } from 'idesign-react';

const App = () => {
  return <Empty image="https://picsum.photos/180/120">自定义展位图</Empty>;
};

export default App;
```

<API />
