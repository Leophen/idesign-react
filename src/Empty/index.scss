$base_width: 200px;
$base_height: auto;

.i-empty {
  display: flex;
  flex-direction: column;
  align-items: center;
  &.i-empty--size-small {
    .i-empty-image {
      width: calc($base_width * 0.8);
    }
    .i-empty-description {
      font-size: 14px;
    }
  }
  &.i-empty--size-large {
    .i-empty-image {
      width: calc($base_width * 1.2);
    }
    .i-empty-description {
      font-size: 18px;
    }
  }
}

.i-empty-image {
  display: flex;
  align-items: flex-end;
  justify-content: center;
  width: $base_width;
  height: $base_height;
  svg {
    width: 100%;
    height: auto;
  }
}

html.dark,
*[data-theme='dark'],
*[data-prefers-color='dark'] {
  .i-empty-image {
    svg {
      filter: invert(1) contrast(0.9) grayscale(1);
    }
  }
}

.i-empty-description {
  margin-top: 20px;
  color: #888;
  font-size: 16px;
}
