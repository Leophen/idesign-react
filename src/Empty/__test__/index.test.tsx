import { render } from '@testing-library/react';
import Empty from '../index';

describe('Empty 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Empty>暂无数据</Empty>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size', () => {
    const { container } = render(<Empty size="small">暂无数据</Empty>);
    expect(container.firstChild).toHaveClass('i-empty--size-small');
  })
});

