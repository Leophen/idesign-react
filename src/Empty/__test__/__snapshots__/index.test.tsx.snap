// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-empty"
  >
    <div
      class="i-empty-image"
    >
      <svg
        fill="none"
        height="420"
        viewBox="0 0 640 420"
        width="640"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_44_427)"
        >
          <path
            d="M321.95 420.27C484.62 420.27 616.49 386.772 616.49 345.45C616.49 304.128 484.62 270.63 321.95 270.63C159.28 270.63 27.4102 304.128 27.4102 345.45C27.4102 386.772 159.28 420.27 321.95 420.27Z"
            fill="url(#paint0_linear_44_427)"
          />
          <path
            d="M153.05 91.3007L315.55 39V183.15L178.98 133.141L153.05 91.3007Z"
            fill="url(#paint1_linear_44_427)"
          />
          <path
            d="M315.55 39L516.65 87.3753L484.95 269.441L315.55 217.469V39Z"
            fill="url(#paint2_linear_44_427)"
          />
          <path
            d="M153.05 91.3008L332.83 144.259V384.82L153.05 321.413V91.3008Z"
            fill="url(#paint3_linear_44_427)"
          />
          <path
            d="M516.65 87.3755V325.338L332.83 384.82V144.26L516.65 87.3755Z"
            fill="url(#paint4_linear_44_427)"
          />
          <path
            d="M268.01 238.06L107.24 170.399L153.05 91.3008L332.83 144.259L268.01 238.06Z"
            fill="url(#paint5_linear_44_427)"
          />
          <path
            d="M516.65 87.3755L582.62 159.621L401.98 238.061L332.83 144.26L516.65 87.3755Z"
            fill="url(#paint6_linear_44_427)"
          />
        </g>
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear_44_427"
            x1="321.954"
            x2="321.954"
            y1="270.635"
            y2="420.274"
          >
            <stop
              stop-color="#EAEAEA"
            />
            <stop
              offset="1"
              stop-color="#FCFCFC"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint1_linear_44_427"
            x1="234.299"
            x2="234.299"
            y1="183.151"
            y2="39.0045"
          >
            <stop
              offset="0.1119"
              stop-color="#DADCE1"
            />
            <stop
              offset="0.9205"
              stop-color="#DFE1E4"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint2_linear_44_427"
            x1="416.096"
            x2="416.096"
            y1="269.442"
            y2="39.0045"
          >
            <stop
              offset="0.0884"
              stop-color="#EBEEF2"
            />
            <stop
              offset="1"
              stop-color="#E8EAEE"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint3_linear_44_427"
            x1="332.832"
            x2="153.052"
            y1="238.064"
            y2="238.064"
          >
            <stop
              offset="0.0884"
              stop-color="#E7E8EC"
            />
            <stop
              offset="1"
              stop-color="#F9F9FA"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint4_linear_44_427"
            x1="344.5"
            x2="536.021"
            y1="392.467"
            y2="100.071"
          >
            <stop
              offset="0.1119"
              stop-color="#DADCE2"
            />
            <stop
              offset="0.9205"
              stop-color="#DADCE2"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint5_linear_44_427"
            x1="332.832"
            x2="107.242"
            y1="164.683"
            y2="164.683"
          >
            <stop
              offset="0.0884"
              stop-color="#F6F7FA"
            />
            <stop
              offset="1"
              stop-color="#F6F7FA"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint6_linear_44_427"
            x1="582.624"
            x2="332.832"
            y1="162.722"
            y2="162.722"
          >
            <stop
              offset="0.0884"
              stop-color="#F6F7FA"
            />
            <stop
              offset="1"
              stop-color="#F6F7FA"
            />
          </lineargradient>
          <clippath
            id="clip0_44_427"
          >
            <rect
              fill="white"
              height="420"
              width="640"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="i-empty-description"
    >
      暂无数据
    </div>
  </div>
</DocumentFragment>
`;
